Snowflake.Data/Logger/SFLoggerImpl.cs

```diff
@@ -0,0 +1,181 @@
+    internal LoggingEvent _level;
+
+    private bool _isDebugEnabled;
+    private bool _isInfoEnabled;
+    private bool _isWarnEnabled;
+    private bool _isErrorEnabled;
+    private bool _isFatalEnabled;
+
+    internal SFLoggerImpl(Type type, LoggingEvent level = LoggingEvent.DEBUG)
+    {
+        _appenders = new List<SFAppender>();
+        _type = type;
+        SetLevel(level);
+    }
+
+    public void SetLevel(LoggingEvent level)
+    {
+        _level = level;
+        SetEnableValues();
+    }
+
+    private void SetEnableValues()
+    {
+        var enabled = _level != LoggingEvent.OFF;
+        _isDebugEnabled = enabled;
+        _isInfoEnabled = enabled;
+        _isWarnEnabled = enabled;
+        _isErrorEnabled = enabled;
+        _isFatalEnabled = enabled;
+
+        if (enabled)
+        {
+            switch (_level)
+            {
+                case LoggingEvent.TRACE:
+                case LoggingEvent.DEBUG:
+                    break;
+                case LoggingEvent.FATAL:
+                    _isErrorEnabled = false;
+                    goto case LoggingEvent.ERROR;
+                case LoggingEvent.ERROR:
+                    _isWarnEnabled = false;
+                    goto case LoggingEvent.WARN;
+                case LoggingEvent.WARN:
+                    _isInfoEnabled = false;
+                    goto case LoggingEvent.INFO;
+                case LoggingEvent.INFO:
+                    _isDebugEnabled = false;
+                    break;
+            }
+        }
+    }
+
+    public List<SFAppender> GetAppenders()
+    {
+        return _appenders;
+    }
+
+    public void AddAppender(SFAppender appender)
+    {
+        _appenders.Add(appender);
+    }
+
+    public void RemoveAppender(SFAppender appender)
+    {
+        _appenders.Remove(appender);
+    }
+
+    public bool IsDebugEnabled()
+    {
+        return SFLogRepository.s_rootLogger == this ?
+            _isDebugEnabled :
+            SFLogRepository.s_rootLogger.IsDebugEnabled();
+    }
+
+    public bool IsInformationEnabled()
+    {
+        return SFLogRepository.s_rootLogger == this ?
+            _isInfoEnabled :
+            SFLogRepository.s_rootLogger.IsInformationEnabled();
+    }
+
+    public bool IsWarningEnabled()
+    {
+        return SFLogRepository.s_rootLogger == this ?
+            _isWarnEnabled :
+            SFLogRepository.s_rootLogger.IsWarningEnabled();
+    }
+
+    public bool IsErrorEnabled()
+    {
+        return SFLogRepository.s_rootLogger == this ?
+            _isErrorEnabled :
+            SFLogRepository.s_rootLogger.IsErrorEnabled();
+    }
+
+    public bool IsFatalEnabled()
+    {
+        return SFLogRepository.s_rootLogger == this ?
+            _isFatalEnabled :
+            SFLogRepository.s_rootLogger.IsFatalEnabled();
+    }
+
+    public void Debug(string msg, Exception ex = null)
+    {
+        if (IsDebugEnabled())
+        {
+            msg = SecretDetector.MaskSecrets(msg).maskedText;
```

Comment:
This way secrets would be masked only in SFLoggerImpl. We need to mask secrets in customer implementation as well.

Reply:
Moved the masking so both loggers use the masked message